spec/chewy/fields/base_spec.rb

```diff
@@ -402,33 +402,279 @@
 
     context 'objects and scopes', :orm do
       before do
+        stub_model(:location)
         stub_model(:city)
         stub_model(:country)
 
         City.belongs_to :country
+        Location.belongs_to :city
+        City.has_one :location, -> { order :id }
         Country.has_many :cities, -> { order :id }
+      end
 
-        stub_index(:countries) do
-          define_type Country do
-            field :id
-            field :cities do
+      context 'text fields type with and without ignore_blank flag' do
+        before do
+          stub_index(:countries) do
+            define_type Country do
               field :id
-              field :name
+              field :cities do
+                field :id
+                field :name
+                field :historical_name, ignore_blank: false
+                field :description, ignore_blank: true
+              end
             end
           end
         end
+
+        let(:country_with_cities) do
+          cities = [
+            City.create!(id: 1, name: '', historical_name: '', description: ''),
+            City.create!(id: 2, name: '', historical_name: '', description: '')
+          ]
+
+          Country.create!(id: 1, cities: cities)
+        end
+
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq(
+            'id' => 1, 'cities' => [
+              {'id' => 1, 'name' => '', 'historical_name' => ''},
+              {'id' => 2, 'name' => '', 'historical_name' => ''}
+            ]
+          )
+        end
       end
 
-      let(:country_with_cities) do
-        cities = [City.create!(id: 1, name: 'City1'), City.create!(id: 2, name: 'City2')]
+      context 'fields with ignore_blank: true flag with .empty? method' do
+        before do
+          stub_index(:countries) do
+            define_type Country do
+              field :id
+              field :cities do
+                field :id
+                field :name, ignore_blank: true
+                field :historical_name, ignore_blank: true
+                field :description, ignore_blank: true
+              end
+            end
+          end
+        end
+
+        let(:country_with_cities) do
+          cities = [
+            City.create!(id: 1, name: '', historical_name: '', description: nil)
+          ]
+
+          Country.create!(id: 1, cities: cities)
+        end
 
-        Country.create!(id: 1, cities: cities)
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq(
+            'id' => 1, 'cities' => [
+              {'id' => 1}
+            ]
+          )
+        end
       end
 
-      specify do
-        expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq('id' => 1, 'cities' => [
-          {'id' => 1, 'name' => 'City1'}, {'id' => 2, 'name' => 'City2'}
-        ])
+      context 'parental and children fields with ignore_blank flag' do
```

Comment:
Is it a parent/child or nested field? The first one reminds me of https://www.elastic.co/guide/en/elasticsearch/reference/6.8/parent-join.html

Reply:
I think, `nested field`.